rotkehlchen/chain/ethereum/modules/curve/curve_cache.py

```diff
@@ -205,171 +213,158 @@ def save_curve_data_to_cache(
         key_parts=[GeneralCacheType.CURVE_LP_TOKENS],
     )
     # Then, read the new pool mappings and save them in the global DB cache
-    globaldb_set_general_cache_values(
-        write_cursor=write_cursor,
-        key_parts=[GeneralCacheType.CURVE_LP_TOKENS],
-        values=pools_mapping.keys(),  # keys of pools_mapping are lp tokens
-    )
-    for lp_token_address, pool_info in pools_mapping.items():
-        pool_address, coins, _underlying_coins = pool_info
+    for pool in all_pools:
         globaldb_set_general_cache_values(
             write_cursor=write_cursor,
-            key_parts=[GeneralCacheType.CURVE_POOL_ADDRESS, lp_token_address],
-            values=[pool_address],
+            key_parts=[GeneralCacheType.CURVE_LP_TOKENS],
+            values=[pool.lp_token_address],  # keys of pools_mapping are lp tokens
         )
         globaldb_set_general_cache_values(
             write_cursor=write_cursor,
-            key_parts=[GeneralCacheType.CURVE_POOL_TOKENS, pool_address],
-            values=coins,
+            key_parts=[GeneralCacheType.CURVE_POOL_ADDRESS, pool.lp_token_address],
+            values=[pool.pool_address],
         )
-    for pool_address, gauge_address in gauges.items():
         globaldb_set_general_cache_values(
             write_cursor=write_cursor,
-            key_parts=[GeneralCacheType.CURVE_GAUGE_ADDRESS, pool_address],
-            values=[gauge_address],
+            key_parts=[GeneralCacheType.CURVE_POOL_TOKENS, pool.pool_address],
+            values=pool.coins,
         )
+        if pool.gauge_address is not None:
+            globaldb_set_general_cache_values(
+                write_cursor=write_cursor,
+                key_parts=[GeneralCacheType.CURVE_GAUGE_ADDRESS, pool.pool_address],
+                values=[pool.gauge_address],
+            )
 
 
-def query_curve_registry_pools(
-        ethereum: 'EthereumInquirer',
-        registry_contract: EvmContract,
-) -> CURVE_POOLS_MAPPING_TYPE:
-    """Query pools from curve registry and return their mappings
-    May raise:
-    - RemoteError if failed to query etherscan or node
-    """
-    registry_pool_count = registry_contract.call(
+def query_curve_data_from_api() -> list[CurvePoolData]:
+    log.debug('Querying curve pools from curve api')
+    all_api_pools = []
+    for api_url in CURVE_API_URLS:
+        try:
+            response = requests.get(api_url, timeout=DEFAULT_TIMEOUT_TUPLE).json()
+        except RequestException as e:
+            raise CurveApiException(f'Could not query curve api due to: {str(e)}') from e
```

Comment:
Your usages(here & below) of `CurveApiException` is simply what `RemoteError` is meant for. 
I don't see a reason to create a new exception class for this.